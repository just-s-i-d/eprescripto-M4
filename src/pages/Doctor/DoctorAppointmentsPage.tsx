import { useEffect, useState } from "react";

import { ColumnsType } from "antd/es/table";
import { Card, Tooltip } from "antd";
import { PhoneTwoTone, CloseCircleTwoTone } from "@ant-design/icons";

import ErrorBoundary from "@components/ErrorBoundary";
import useStatesHook from "../../hooks/useStatesHook";

import {
  cancelAppointment,
  formatDateReadable,
  getDoctorAppointmentsData,
} from "@utils/Doctor";
import {
  AppointmentDataType,
  DoctorAppointmentsDataType,
} from "@constants/types";
import TableCard from "@components/ui/TableCard";
import PopModal from "@components/ui/PopModal";
import { showToast } from "@utils/common";

const DoctorAppointmentsPage = () => {
  const appointmentsTable = useStatesHook<DoctorAppointmentsDataType>();
  const [open, setOpen] = useState(false);
  const [selectedPatient, setSelectedPatient] = useState<AppointmentDataType>();
  const handlerCancleAppointment = (patient: AppointmentDataType) => {
    setOpen(true);
    setSelectedPatient(patient);
  };
  const confirmHandler = async () => {
    if (selectedPatient) {
      const response = await cancelAppointment(selectedPatient);
      response && showToast(response.message, response.type);
    }
  };
  useEffect(() => {
    getDoctorAppointmentsData()
      .then((res) => {
        appointmentsTable.setData(res);
        appointmentsTable.setError(false);
        appointmentsTable.setLoading(false);
      })
      .catch(() => {
        appointmentsTable.setError(true);
        appointmentsTable.setLoading(false);
      });
  }, []);
  const doctorAppointmentsTableColumns: ColumnsType<AppointmentDataType> = [
    {
      title: "Patient Id",
      dataIndex: "pId",
      width: 50,
    },
    {
      title: "Patient Name",
      dataIndex: "pName",
      width: 150,
    },
    {
      title: "Email",
      dataIndex: "email",
      width: 200,
    },
    {
      title: "Date",
      dataIndex: "date",
      width: 150,
      render: (date, record) => (
        <div className="flex flex-col">
          <span>{formatDateReadable(date)}</span>
          <span>{record.timeSlot}</span>
        </div>
      ),
    },
    {
      title: "Referrer",
      dataIndex: "referer",
      width: 100,
    },
    {
      title: "Status",
      dataIndex: "status",
      width: 50,
      render: (value: string) => (
        <span className={`py-1 px-2 rounded-lg ${value.toLowerCase()}`}>
          {value}
        </span>
      ),
      filters: [
        { text: "Pending", value: "Pending" },
        { text: "Attended", value: "Attended" },
        { text: "Cancelled", value: "Cancelled" },
      ],
      onFilter: (value, record) => record.status === value,
    },
    {
      title: "Action",
      width: 100,
      render: (_, record) => (
        <div className="flex gap-4 text-lg">
          <Tooltip title={record.contact}>
            <PhoneTwoTone
              className="rotate-90 cursor-pointer fill-[#19504c]"
              href={`tel:${record.contact}`}
            />
          </Tooltip>
          {record.status === "Pending" && (
            <CloseCircleTwoTone
              className="cursor-pointer"
              onClick={() => handlerCancleAppointment(record)}
            />
          )}
        </div>
      ),
    },
  ];
  return (
    <>
      <ErrorBoundary>
        <Card className="appointment-table-card h-min w-full overflow-x-scroll">
          <TableCard
            className="min-w-[1100px]"
            columns={doctorAppointmentsTableColumns}
            tableData={appointmentsTable.data}
            pageSize={8}
          />
          <PopModal
            setOpen={setOpen}
            open={open}
            okButtonText="Sure"
            title="Cancel Appointment"
            footer={true}
            confirmHandler={confirmHandler}
          >
            Are you sure,you want to cancel the appoinment for{" "}
            {selectedPatient?.pName}
          </PopModal>
        </Card>
      </ErrorBoundary>
    </>
  );
};
export default DoctorAppointmentsPage;
