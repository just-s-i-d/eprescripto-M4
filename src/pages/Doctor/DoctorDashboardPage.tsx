import { useEffect } from "react";

import NextPatientCard from "@components/Doctor/NextPatientCard";
import InfoCards from "@components/Doctor/InfoCards";
import ErrorBoundary from "@components/ErrorBoundary";
import LineChart from "@components/Charts/LineChart";
import BarChart from "@components/Charts/BarChart";
import AppoinmentsCard from "@components/Doctor/AppoinmentsCard";
import DoughnutChart from "@components/Charts/DoughnutChart";
import {
  appointmentsEndPoint,
  barChartDataEndPoint,
  doughnutChartDataEndPoint,
  getData,
  getDataForLineGraph,
  infoCardDataEndPoint,
  lineChartDataEndPoint,
} from "@utils/Doctor";
import useStatesHook from "../../hooks/useStatesHook";
import {
  ApiResponseData,
  ApiResponseDataType,
  AppointmentDataType,
  ChartDataType,
  InfoCardDetailsType,
  LineChartApiResDataType,
} from "@constants/types";

const DoctorDashboardPage: React.FC = () => {
  const infoCards = useStatesHook<ApiResponseData<InfoCardDetailsType>>();
  const appointments = useStatesHook<ApiResponseData<AppointmentDataType>>();
  const doughnutChart = useStatesHook<ApiResponseDataType<ChartDataType>>();
  const barChart = useStatesHook<ApiResponseDataType<ChartDataType>>();
  const lineChart =
    useStatesHook<ApiResponseDataType<LineChartApiResDataType>>();
  useEffect(() => {
    getData(infoCardDataEndPoint)
      .then((res) => {
        infoCards.setData(res);
        infoCards.setError(false);
      })
      .catch(() => {
        infoCards.setError(true);
      });
  }, [infoCards.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(appointmentsEndPoint)
      .then((res) => {
        appointments.setData(res);
        appointments.setLoading(false);
        appointments.setError(false);
      })
      .catch(() => {
        appointments.setError(true);
        appointments.setLoading(false);
      });
  }, [appointments.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(doughnutChartDataEndPoint)
      .then((res) => {
        doughnutChart.setData(res);
        doughnutChart.setLoading(false);
        doughnutChart.setError(false);
      })
      .catch(() => {
        doughnutChart.setError(true);
      });
  }, [doughnutChart.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(lineChartDataEndPoint)
      .then((res) => {
        lineChart.setData(res);
        lineChart.setError(false);
      })
      .catch(() => {
        lineChart.setError(true);
      });
  }, [lineChart.refresh]); //eslint-disable-line

  useEffect(() => {
    getData(barChartDataEndPoint)
      .then((res) => {
        barChart.setData(res);
        barChart.setError(false);
      })
      .catch(() => {
        barChart.setError(true);
      });
  }, [barChart.refresh]); //eslint-disable-line

  return (
    <>
      <ErrorBoundary>
        <div className="max-w-full flex">
          <div className="w-full flex flex-col gap-6">
            <ErrorBoundary>
              <InfoCards cardDetails={infoCards.data} infoCards={infoCards} />
            </ErrorBoundary>
            <div className="flex flex-wrap justify-between gap-y-6 max-tablet:justify-center">
              <ErrorBoundary>
                <NextPatientCard
                  appointments={appointments}
                  appointment={appointments.data && appointments.data[0]}
                  error={appointments.error}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <DoughnutChart
                  doughnutChart={doughnutChart}
                  chartData={doughnutChart.data}
                  title="Ratings Composition"
                  label="Reviews"
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <AppoinmentsCard
                  appointmentsState={appointments}
                  appointments={appointments.data}
                  loading={appointments.loading}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <LineChart
                  lineChart={lineChart}
                  chartData={lineChart.data}
                  title="Patients in Last 7 Days"
                  xAxesTitle="Dates"
                  yAxesTitle="Number of Patients"
                  getDataForLineGraph={getDataForLineGraph}
                />
              </ErrorBoundary>
              <ErrorBoundary>
                <BarChart
                  barChart={barChart}
                  chartData={barChart.data}
                  title="Age Groups Comparison"
                  label="Patients"
                  xAxesTitle="Different Age Groups (In Years)"
                  yAxesTitle="Number of Patients"
                />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </ErrorBoundary>
    </>
  );
};

export default DoctorDashboardPage;
