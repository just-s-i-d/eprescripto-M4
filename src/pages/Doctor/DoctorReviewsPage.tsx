import { useEffect } from "react";

import { Card, Table } from "antd";
import { ColumnsType } from "antd/es/table";

import { ApiResponseData, ReviewDataType } from "@constants/types";
import { getData, reviewsDataEndPoint } from "@utils/Doctor";
import useStatesHook from "../../hooks/useStatesHook";
import starPng from "@assets/star.png";
import ErrorBoundary from "@components/ErrorBoundary";

const DoctorReviewsPage = () => {
  const reviews = useStatesHook<Record<string, string | number>[]>();
  const columns: ColumnsType<ReviewDataType> = [
    {
      title: "S. No",
      render: (_, _1, index) => index + 1,
    },
    {
      title: "Rating",
      dataIndex: "rating",
      key: "rating",
      render: (rating: number) => (
        <span className="flex">
          {Array.from({ length: rating }, (_, index) => (
            <img
              src={starPng}
              key={index}
              role="img"
              aria-label="star"
              width={20}
            />
          ))}
        </span>
      ),
      filters: [
        {
          text: "1 Star",
          value: 1,
        },
        {
          text: "2 Star",
          value: 2,
        },
        {
          text: "3 Star",
          value: 3,
        },
        {
          text: "4 Star",
          value: 4,
        },
        {
          text: "5 Star",
          value: 5,
        },
      ],
      onFilter: (value, record) => record.rating === value,
      sorter: {
        compare: (a, b) => a.rating - b.rating,
        multiple: 3,
      },
    },
    {
      title: "Comment",
      dataIndex: "comment",
      key: "comment",
    },
  ];
  useEffect(() => {
    getData<ApiResponseData<Record<string, string | number>>>(
      reviewsDataEndPoint,
    )
      .then((res) => {
        const data = res.map((element) => element.attributes);
        reviews.setData(data);
        reviews.setError(false);
        reviews.setLoading(false);
      })
      .catch(() => {
        reviews.setError(true);
        reviews.setLoading(false);
      });
  }, [reviews.refresh]);
  return (
    <>
      <Card className="overflow-scroll min-h-[50vh]">
        <ErrorBoundary
          error={reviews.error}
          refreshComponent={() => reviews.setRefresh((prev) => !prev)}
        >
          <Table
            className="custom-table min-w-[800px]"
            columns={columns as ColumnsType<Record<string, string | number>>}
            dataSource={reviews.data}
            loading={reviews.loading}
          />
        </ErrorBoundary>
      </Card>
    </>
  );
};
export default DoctorReviewsPage;
