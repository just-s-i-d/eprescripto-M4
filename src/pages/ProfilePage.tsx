import ErrorBoundary from "@components/ErrorBoundary";
import PersonalInfoCard from "@components/Profile/PersonalInfoCard";
import ProfessionalInfoCard from "@components/Profile/ProfessionalInfoCard";
import { getUserData } from "@utils/Doctor";
import { Space } from "antd";
import { useEffect, useState } from "react";

const ProfilePage = () => {
  const [userData, setUserData] = useState();
  useEffect(() => {
    getUserData().then((res) => {
      setUserData(res);
    });
  }, []);
  return (
    <ErrorBoundary>
      <Space direction="vertical" size={20}>
        <PersonalInfoCard userData={userData} />
        <ProfessionalInfoCard userData={userData} />
      </Space>
    </ErrorBoundary>
  );
};
export default ProfilePage;
