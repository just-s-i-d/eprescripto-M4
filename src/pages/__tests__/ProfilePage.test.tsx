import ProfilePage from "@pages/ProfilePage";
import { render, waitFor, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@test/matchMedia";
import { deleteAccount } from "@utils/Doctor";

jest.mock("@utils/Doctor", () => ({
  getUserData: jest.fn().mockResolvedValue({
    docName: "John Doe",
    email: "john.doe@example.com",
    dob: "1990-01-01",
    gender: "male",
    city: "City",
    state: "State",
    country: "Country",
    profilePic: "profilePic.jpg",
    licenseNo: 1234,
    speciality: "Cardiology",
    experience: 5,
    organizationType: "hospital",
    organizationName: "ABC Hospital",
  }),
  deleteAccount: jest.fn(),
}));

describe("Profile Page Tests", () => {
  it("Personal info and Professional info card render test", () => {
    const { asFragment } = render(<ProfilePage darkMode={false} />);
    expect(screen.getByText("Personal Information")).toBeInTheDocument();
    expect(screen.getByText("Professional Information")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("delete btn render test", async () => {
    render(<ProfilePage darkMode={false} />);
    const deleteAccountBtn = screen.getByRole("button", {
      name: "Delete Account",
    });
    expect(deleteAccountBtn).toBeInTheDocument();
  });

  it("Account delete test", async () => {
    render(<ProfilePage darkMode={false} />);

    const deleteAccountBtn = screen.getByRole("button", {
      name: "Delete Account",
    });
    expect(deleteAccountBtn).toBeInTheDocument();
    fireEvent.click(deleteAccountBtn);
    const confirmBtn = screen.getByRole("button", { name: "Yes" });
    expect(confirmBtn).toBeInTheDocument();
    fireEvent.click(confirmBtn);
    await waitFor(() => {
      expect(deleteAccount).toHaveBeenCalled();
    });
  });
});
