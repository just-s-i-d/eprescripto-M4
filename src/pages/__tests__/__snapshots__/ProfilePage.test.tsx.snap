// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Tests Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="ant-space css-dev-only-do-not-override-1adbn6x ant-space-vertical"
    style="column-gap: 20px; row-gap: 20px;"
  >
    <div
      class="ant-space-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-card-body"
        >
          <h1
            class="text-xl font-bold mb-6 undefined"
          >
            Personal Information
          </h1>
          <div>
            <form
              class="ant-form ant-form-vertical css-dev-only-do-not-override-1adbn6x custom-form custom-form-profile w-full flex xs:flex-col xs:items-start sm:items-center sm:justify-between mb-4  sm:flex-row-reverse"
            >
              <div
                class="flex justify-center xs:w-full sm:w-4/12"
              >
                <div
                  class="ant-form-item css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            accept="image/*"
                            class="ant-input css-dev-only-do-not-override-1adbn6x hidden"
                            id="imageInput"
                            type="file"
                            value=""
                          />
                          <span
                            class="flex flex-col items-center justify-center"
                          >
                            <img
                              alt="Profile Preview"
                              class="xs:w-[120px] md:w-[150px] rounded-[50%] mb-4"
                              src="profilepic.png"
                            />
                            <button
                              class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default w-full"
                              disabled=""
                              type="button"
                            >
                              <label
                                for="imageInput"
                              >
                                Change Picture
                              </label>
                            </button>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-col gap-3 xs:w-full sm:w-7/12 md:w-9/12 xl:w-8/12"
              >
                <div
                  class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
                >
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="docName"
                          title="Full Name"
                        >
                          Full Name
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                              disabled=""
                              id="docName"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="email"
                          title="E-mail"
                        >
                          E-mail
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                              disabled=""
                              id="email"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
                >
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="dob"
                          title="Date of birth"
                        >
                          Date of birth
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                              disabled=""
                              id="dob"
                              type="date"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="gender"
                          title="Gender"
                        >
                          Gender
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <div
                              aria-required="true"
                              class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow ant-select-disabled"
                            >
                              <div
                                class="ant-select-selector"
                              >
                                <span
                                  class="ant-select-selection-search"
                                >
                                  <input
                                    aria-autocomplete="list"
                                    aria-controls="gender_list"
                                    aria-expanded="false"
                                    aria-haspopup="listbox"
                                    aria-owns="gender_list"
                                    aria-required="true"
                                    autocomplete="off"
                                    class="ant-select-selection-search-input"
                                    disabled=""
                                    id="gender"
                                    readonly=""
                                    role="combobox"
                                    style="opacity: 0;"
                                    type="search"
                                    unselectable="on"
                                    value=""
                                  />
                                </span>
                                <span
                                  class="ant-select-selection-placeholder"
                                >
                                  Select your gender
                                </span>
                              </div>
                              <span
                                aria-hidden="true"
                                class="ant-select-arrow"
                                style="user-select: none;"
                                unselectable="on"
                              >
                                <span
                                  aria-label="down"
                                  class="anticon anticon-down ant-select-suffix"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="flex xs:gap-3 md:gap-6 xs:flex-col md:flex-row"
                >
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="city"
                          title="City"
                        >
                          City
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                              disabled=""
                              id="city"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                      >
                        <label
                          class="ant-form-item-required"
                          for="state"
                          title="State"
                        >
                          State
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                              disabled=""
                              id="state"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-form-item xs:w-full md:w-[48%] xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                    >
                      <label
                        class="ant-form-item-required"
                        for="country"
                        title="Country"
                      >
                        Country
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            aria-required="true"
                            class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                            disabled=""
                            id="country"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <div
              class="flex xs:gap-3 md:gap-6"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
                type="button"
              >
                <span>
                  Edit Details
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-card-body"
        >
          <h1
            class="text-xl font-bold mb-6 undefined"
          >
            Professional Information
          </h1>
          <form
            class="ant-form ant-form-vertical css-dev-only-do-not-override-1adbn6x custom-form custom-form-profile w-full flex flex-col gap-2 mb-4"
          >
            <div
              class="flex xs:flex-col sm:flex-row xs:gap-2 sm:gap-6 sm:w-full md:w-9/12 xl:w-8/12"
            >
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="licenseNo"
                      title="License Number"
                    >
                      License Number
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="licenseNo"
                          type="number"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="speciality"
                      title="Speciality"
                    >
                      Speciality
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="speciality"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex xs:flex-col sm:flex-row xs:gap-2 sm:gap-6 sm:w-full md:w-9/12 xl:w-8/12"
            >
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="experience"
                      title="Experience"
                    >
                      Experience
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                          disabled=""
                          id="experience"
                          type="number"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item xs:w-full xl:w-5/12 css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                  >
                    <label
                      class="ant-form-item-required"
                      for="organizationType"
                      title="Organization Type"
                    >
                      Organization Type
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          aria-required="true"
                          class="ant-select ant-select-in-form-item css-dev-only-do-not-override-1adbn6x ant-select-single ant-select-show-arrow ant-select-disabled"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-autocomplete="list"
                                aria-controls="organizationType_list"
                                aria-expanded="false"
                                aria-haspopup="listbox"
                                aria-owns="organizationType_list"
                                aria-required="true"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                disabled=""
                                id="organizationType"
                                readonly=""
                                role="combobox"
                                style="opacity: 0;"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-item"
                              title="Select an option"
                            >
                              Select an option
                            </span>
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item xs:full sm:w-[49%] md:w-[36%] xl:w-[28%] css-dev-only-do-not-override-1adbn6x"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-1adbn6x"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-1adbn6x"
                >
                  <label
                    class="ant-form-item-required"
                    for="organizationName"
                    title="Organization Name"
                  >
                    Organization Name
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-1adbn6x"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input ant-input-disabled css-dev-only-do-not-override-1adbn6x"
                        disabled=""
                        id="organizationName"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
          <div
            class="ant-space css-dev-only-do-not-override-1adbn6x ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
          >
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default"
                type="button"
              >
                <span>
                  Edit Details
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-1adbn6x"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="flex items-center justify-between gap-6"
          >
            <span
              class="font-extrabold sm:text-base"
            >
              Once you delete your account, there is no going back.
            </span>
            <button
              class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-default ant-btn-dangerous w-[150px]"
              type="button"
            >
              <span>
                Delete Account
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
